lib/cadet/stories/assets.ex

```diff
@@ -0,0 +1,66 @@
+defmodule Cadet.Stories.Assets do
+  @moduledoc """
+  Assessments context contains domain logic for assets management
+  for Source academy's game component
+  """
+  @manage_assets_role ~w(staff admin)a
+
+  @accessible_folders ~w(images locations objects avatars ui stories testFolder)
+  @accepted_file_types ~w(.jpg .jpeg .gif .JPG .txt .jpeg .wav .mp3 .png)
+
+  @bucket_name "source-academy-assets"
+  @fetch_limit 2000
+
+  def validate_assets_role(conn) do
+    if conn.assigns[:current_user].role in @manage_assets_role do
+      :ok
+    else
+      {:error, {:forbidden, "User not allowed to manage assets"}}
+    end
+  end
+
+  def validate_folder_name(folder_name) do
+    if Enum.member?(@accessible_folders, folder_name) do
+      :ok
+    else
+      {:error, {:bad_request, "Bad Request"}}
+    end
+  end
+
+  def validate_filetype(filename) do
```

Comment:
`filename` seems like it should be `filetype` or `fileext` or something like that.